temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowTaskStateMachine.java

```diff
@@ -108,30 +109,29 @@ private WorkflowTaskStateMachine(long workflowTaskStartedEventId, Listener liste
                   WorkflowTaskStateMachine::handleFailed)
               .add(State.STARTED, EventType.EVENT_TYPE_WORKFLOW_TASK_TIMED_OUT, State.TIMED_OUT);
 
+  private synchronized void startWorkflowTaskOnce(boolean lastTaskInHistory) {
```

Comment:
State machine code doesn't run concurrently so you don't need this synchronization.